### PR TITLE
MHV Secondary Nav can have links outside an app

### PR DESCRIPTION
**Pull request by carlosfelix2**
## Summary
This change allows the MHV secondary nav to have item links that point to a location outside an app's root URL. This is done by the addition of a new property called `appRootUrl` that can be different from the `href` property. For example, we want the Medications link in the secondary nav to point to `/my-health/medications/about` while Medications is being deployed, but we still want the secondary nav to highlight the Medications item when the URL is within the app's root URL such as `/my-health/medications`. This change also allows the secondary nav link to be completely outside of an app's root URL, such as when we use the temporary medical records page of `/my-health/records` while the Medical Records app root URL is `/my-health/medical-records`. 

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/83318

## Testing done
- Updated unit tests

### Manual testing
1. Run the landing page locally by running `yarn watch --env --entry=mhv-landing-page`
2. Change the local mocks on the last line of the file `src/applications/mhv-landing-page/mocks/api/index.js` to use `MOCK_TYPES.VERIFIED_USER_ALL_FEATURES`
3. Run the local mock by running `yarn mock-api --responses src/applications/mhv-landing-page/mocks/api/index.js`
4. Access the landing page at http://localhost:3001/my-health/
5. Click on the Medications link and verify it takes you to the Medications about page and that the secondary nav Medications item is highlighted.
6. Navigate to another Medications page like http://localhost:3001/my-health/medications and verify that the secondary nav Medications item is highlighted.

## Screenshots
No changes to the UI made.

## What areas of the site does it impact?
- MHV secondary nav shared component

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

### CODE REVIEW

File: src/platform/mhv/secondary-nav/tests/MhvSecondaryNavMenu.unit.spec.jsx [hunk 1]
```diff
@@ -25,71 +25,139 @@ const testSecNavItems = [
...
-    const { getAllByRole } = render(
-      <Provider store={mock}>
-        <MhvSecondaryNavMenu items={testSecNavItems} />
-      </Provider>,
-    );
-    const links = getAllByRole('link');
-    expect(links.length).to.eql(testSecNavItems.length);
-    testSecNavItems.forEach((navItem, i) => {
-      expect(links[i].pathname).to.be.eql(navItem.href);
+  describe('feature toggle', () => {
```
Comment (carlosfelix2) [inline src/platform/mhv/secondary-nav/tests/MhvSecondaryNavMenu.unit.spec.jsx:59]: Note: Only indentation changes until line 88. I added another describe block.

File: src/platform/mhv/secondary-nav/tests/MhvSecondaryNavMenu.unit.spec.jsx [hunk 2]
```diff
@@ -25,71 +25,139 @@ const testSecNavItems = [
...
-      const { getAllByTestId } = render(
+    const medNavItem = {
+      title: 'Medications',
+      abbreviation: 'Meds',
+      icon: 'fas fa-prescription-bottle',
+      href: '/my-health/records',
+      appRootUrl: '/my-health/medications',
+    };
+
+    it('based on href', () => {
```
Comment (carlosfelix2) [inline src/platform/mhv/secondary-nav/tests/MhvSecondaryNavMenu.unit.spec.jsx:96]: Note: This test is unchanged.

File: src/platform/mhv/secondary-nav/components/MhvSecondaryNavMenu.jsx [hunk 3]
```diff
@@ -7,12 +7,43 @@ import MhvSecondaryNavItem from './MhvSecondaryNavItem';
...
+   * Find which navigation item needs to be set to active, if any. An item should be active
+   * when the URL pathname starts with the app's root URL, or the href matches the current
+   * URL pathname.
+   * @param secNavItems the list of navigation items
+   * @returns the item to be set as active, or null if none found
+   */
+  const findActiveItem = (secNavItems = items) => {
+    // Perform a reverse find to match which nav link we are on, so we match on the home page last
+    return [...secNavItems] // Clone the array, so the original stays the same
+      .reverse()
```
Comment (dcloud) [inline src/platform/mhv/secondary-nav/components/MhvSecondaryNavMenu.jsx:34]: I take it `toReversed()` isn't in our babel polyfills yet… interesting that the non-mutating method is so new

**Review by dcloud — APPROVED** [review]
Some of the changes in mocks/api looks familiar… I guess it was in another PR that hasn't been merged?

This looks good. I ran `yarn watch --env entry=mhv-landing-page,auth,medications,vaos,medical-records,mhv-secure-messaging` and the mock API with most feature flags enabled, and I see the Medications "About" page with the selection in secondary nav.

**Comment by carlosfelix2** [comment]
That was merged. I added a couple of things to it.
> Some of the changes in mocks/api looks familiar… I guess it was in another PR that hasn't been merged?
> 
> This looks good. I ran `yarn watch --env entry=mhv-landing-page,auth,medications,vaos,medical-records,mhv-secure-messaging` and the mock API with most feature flags enabled, and I see the Medications "About" page with the selection in secondary nav.

**Pull request merged**